graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/migration/CassandraSchemaHelper.java

```diff
@@ -109,12 +115,63 @@ private static boolean equals(Column.ColumnType expectedType, Column.ColumnType
     return true;
   }
 
-  /** @return a list of differences, or empty if the UDTs match. */
-  public static List<Difference> compare(UserDefinedType expectedType, UserDefinedType actualType) {
+  private static SecondaryIndex findIndex(Table table, Column column) {
+    if (column == null) {
+      return null;
+    }
+    return (SecondaryIndex)
+        table.indexes().stream()
+            .filter(
+                index ->
+                    index instanceof SecondaryIndex
+                        && column.equals(((SecondaryIndex) index).column()))
+            .findFirst()
+            .orElse(null);
+  }
+
+  private static void compareIndex(
+      SecondaryIndex expectedIndex,
+      SecondaryIndex actualIndex,
+      Column expectedColumn,
+      List<Difference> differences) {
+
+    if (actualIndex == null) {
+      differences.add(
+          new Difference(expectedColumn, expectedIndex, DifferenceType.MISSING_INDEX, null));
+      return;
```

Comment:
what if we have > 1 missing indexes? would it find only the 1st one and return it?

Reply:
This is for a single index comparison. There's nothing left to check here if the actual index is missing it'll add the difference and the loop where this is called will keep looping for other columns/indexes.

Reply:
Yes, this is in the context of a single column so there can be at most 1 index missing.

However I just realized that CQL _does_ allow more than one index per column, for a map column you can have both KEYS, VALUES and ENTRIES indexes at the same time. We don't support maps at the moment, but for the sake of future compatibility it might be better to have a separate `@cql_index` annotation.